src/serialization.cpp

```diff
@@ -69,22 +69,22 @@ Status serializeBlobToTensorProto(
     }
     }
     responseOutput.mutable_tensor_shape()->Clear();
-    auto& effectiveNetworkOutputShape = networkOutput->getEffectiveShape();
+    auto& effectiveNetworkOutputShape = networkOutput->getShape_3();
     auto& actualBlobShape = getEffectiveBlobShape(blob);
     if (effectiveNetworkOutputShape.size() != actualBlobShape.size()) {
         SPDLOG_ERROR("Failed to serialize blob: {}. There is difference in number of dimensions expected:{} vs actual:{}",
             networkOutput->getName(), effectiveNetworkOutputShape.size(), actualBlobShape.size());
         return StatusCode::INTERNAL_ERROR;
     }
     for (size_t i = 0; i < effectiveNetworkOutputShape.size(); ++i) {
-        size_t dim;
-        if (effectiveNetworkOutputShape[i] != 0) {
-            if (effectiveNetworkOutputShape[i] != actualBlobShape[i]) {
+        dimension_value_t dim;
+        if (effectiveNetworkOutputShape[i] != DYNAMIC_DIMENSION) {
+            if (!effectiveNetworkOutputShape[i].match(actualBlobShape[i])) {
                 SPDLOG_ERROR("Failed to serialize blob: {}. There is difference in dimension:{} expected:{} vs actual:{}",
-                    networkOutput->getName(), i, effectiveNetworkOutputShape[i], actualBlobShape[i]);
+                    networkOutput->getName(), i, effectiveNetworkOutputShape[i].toString(), actualBlobShape[i]);
                 return StatusCode::INTERNAL_ERROR;
             }
-            dim = effectiveNetworkOutputShape[i];
+            dim = effectiveNetworkOutputShape[i].getAnyValue();
```

Comment:
`getAnyValue` and `getFlatShape` will be removed during cleanup phase. Use `isStatic`/`isDynamic` to decide what to do in both situations.

Reply:
This is dead code to be removed in cleanup.